### PR TITLE
[#295] support envVar[] in the CR

### PR DESCRIPTION
**Pull request by gtully**

### CODE REVIEW

**Review by gaohoward — APPROVED** [review]

**Comment by gaohoward** [comment]
@gtully @brusdev @RoddieKieley We should probably bump the crd version first? i.e. merge this pr befor that:
https://github.com/artemiscloud/activemq-artemis-operator/pull/252

**Comment by gtully** [comment]
For additions, we can stick with the CRD version. We really only have to bump when we are breaking existing CRs.

**Comment by gaohoward** [comment]
@gtully @brusdev @RoddieKieley ok. I'd then suggest we cut a new tag before merging any additions.

File: controllers/activemqartemis_reconciler.go [hunk 1]
```diff
@@ -1633,13 +1631,12 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
...
 	// this depends on init container passing --java-opts to artemis create via launch.sh *and* it
-	// not getting munged on the way.
-	// REVISIT: should an existing value be respected here, only when we expose JAVA_OPTS in the CR?
+	// not getting munged on the way. We CreateOrAppend to any value from spec.Env
 	javaOpts := corev1.EnvVar{
 		Name:  "JAVA_OPTS",
 		Value: "-Dbroker.properties=/amq/extra/configmaps/" + brokerPropertiesConfigMapName + "/broker.properties",
 	}
-	environments.Create(podSpec.InitContainers, &javaOpts)
+	environments.CreateOrAppend(podSpec.InitContainers, &javaOpts)
```
Comment (brusdev) [inline controllers/activemqartemis_reconciler.go:1639]: as final user maybe I would expect see my env vars with the same value defined in the CR
as operator maybe I should avoid to change my env vars
a solution could be that the operator ignore internal vars as JAVA_OPTS and include a prefix/suffix env var in the internal env vars. i.e. JAVA_OPTS="$JAVA_OPTS_PRE ... $JAVA_OPTS_SUF"
so users are not able to set JAVA_OPTS but they are able to set $JAVA_OPTS_PRE or $JAVA_OPTS_SUF

Comment (gtully) [reply controllers/activemqartemis_reconciler.go:1639]: I think JAVA_OPTS is some what of a special case b/c it is interpreted by the init container, artemis create command and is used for brokerProperties. I think we can in the future configure the brokerProperties url in some better way and then not have to treat is specially w.r.t append. 
But for sure I don't want to encourage the use of env JAVA_OPTS. it is particular to the artemis launch scripts... way too low level. Ading PRE POST special cases would make it worse imho.
To configure the JVM, use JDK_JAVA_OPTIONS

**Review by brusdev — APPROVED** [review]

**Comment by gtully** [comment]
going to add some doc, and noticed the name annotation is wrong!

**Pull request merged**